ddtrace/llmobs/experimentation/_config.py

```diff
@@ -0,0 +1,196 @@
+) -> None:
+    """
+    Initialize an experiment environment for pushing and retrieving data from Datadog.
+
+    This function configures global settings needed to interact with the Datadog LLM Observability API.
+    If 'run_locally' is True, no data will be sent; otherwise, this function ensures that all required
+    credentials are available. If no credentials are passed in, it attempts to read them from environment
+    variables.
+
+    Args:
+        ml_app (str): The ML application name.
+        project_name (str): Name of the project to be associated with any experiments.
+        site (str, optional): Datadog site domain (e.g., "datadoghq.com"). Defaults to "datadoghq.com".
+        api_key (str, optional): Datadog API key. Reads from environment variable DD_API_KEY if None.
+        application_key (str, optional): Datadog Application key. Reads from environment variable DD_APPLICATION_KEY if None.
+        run_locally (bool, optional): If True, disables communication with Datadog and runs locally.
+
+    Raises:
+        ConfigurationError: If required environment variables or parameters are missing or invalid when run_locally is False.
+    """
+    from .utils._ui import Color 
+
+    global _IS_INITIALIZED, _ENV_PROJECT_NAME, _ENV_DD_SITE, _ENV_DD_API_KEY, \
+           _ENV_DD_APPLICATION_KEY, _RUN_LOCALLY, _ML_APP
+
+    if run_locally:
+        _RUN_LOCALLY = True
+        return
+
+    if api_key is None:
+        api_key = os.getenv("DD_API_KEY")
+        if api_key is None:
+            raise ConfigurationError(
+                "DD_API_KEY environment variable is not set, please set it or pass it as an argument to init(...api_key=...)"
+            )
+
+    if application_key is None:
+        application_key = os.getenv("DD_APPLICATION_KEY")
+        if application_key is None:
+            raise ConfigurationError(
+                "DD_APPLICATION_KEY environment variable is not set, please set it or pass it as an argument to init(...application_key=...)"
+            )
+
+    if site is None:
+        site = os.getenv("DD_SITE")
+        if site is None:
+             raise ConfigurationError(
+                 "DD_SITE environment variable is not set, please set it or pass it as an argument to init(...site=...)"
+             )
+
+    if site not in VALID_DD_SITES:
+        raise ConfigurationError(
+            f"Invalid Datadog site '{site}' provided. Must be one of: {', '.join(sorted(VALID_DD_SITES))}"
+        )
+
+
+    if not _IS_INITIALIZED:
+        trace_enabled_env = os.getenv("DD_TRACE_ENABLED")
+        if trace_enabled_env is None:
+            os.environ["DD_TRACE_ENABLED"] = "false"
+            print(
+                f"{Color.YELLOW}Warning: APM tracing is disabled by default for LLM Experiments. Only LLM Observability tracing is enabled. "
+                f"To enable APM tracing, set the DD_TRACE_ENABLED=true environment variable.{Color.RESET}"
+            )
+
+
+        LLMObs.enable(
```

Comment:
I'd really like to avoid calling enable again. This is broken if the customer already has an `enable()` call in their code from manually using llmobs which means any settings defined here will be ignored.